apps/meteor/server/lib/auditServerEvents/userChanged.ts

```diff
@@ -0,0 +1,168 @@
+import type { IAuditServerUserActor, IServerEvents, ExtractDataToParams, IUser } from '@rocket.chat/core-typings';
+import { ServerEvents } from '@rocket.chat/models';
+import type { UpdateFilter } from 'mongodb';
+
+const userKeysToObfuscate = ['authorizedClients', 'e2e', 'inviteToken', 'oauth'];
+const nestableKeysToObfuscate = ['services', 'password', 'bcrypt']; // ex: services.password.bcrypt
+
+const obfuscateServices = (services: Record<string, any>): Record<string, any> => {
+	return Object.fromEntries(
+		Object.keys(services).map((key) => {
+			// Email 2FA is okay, only tells if it's enabled
+			if (key === 'email2fa') {
```

Comment:
For future reference, once we started auditing users created/updated from external services, we might want to expand this with an extended list of useful information from the services that shouldn't be censored out.